rclpy/rclpy/publisher.py

```diff
@@ -51,6 +58,23 @@ def __init__(
         self.topic = topic
         self.qos_profile = qos_profile
         self.node_handle = node_handle
+        self.callback_group = callback_group
+
+        self.event_callbacks = event_callbacks
+        self.event_handlers: List[QoSEventHandler] = []
+        if event_callbacks:
+            if event_callbacks.deadline:
```

Comment:
Sounds like this could be split into multiple functions, this starts to be a bit long

Reply:
Going to implement as function on EventCallbacks to keep centralized - as per Nick's comment below